File: app/src/main/java/com/memory_athlete/memoryassistant/disciplines/Faces.java
package com.memory_athlete.memoryassistant.disciplines;

import android.os.Bundle;
import android.support.annotation.NonNull;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.EditText;

import com.memory_athlete.memoryassistant.R;
import com.memory_athlete.memoryassistant.recall.RecallComplex;

import java.io.File;
import java.util.ArrayList;
import java.util.Random;

public class Faces extends ComplexDisciplineFragment {
    String[] faces;
    ArrayList<Integer> randomList;

    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        super.onCreateView(inflater, container, savedInstanceState);
        ((EditText) rootView.findViewById(R.id.no_of_values))
                .setHint(getString(R.string.enter) + " " + getString(R.string.images));

        mRecallClass = RecallComplex.class;
        hasSpeech = false;
        rootView.findViewById(R.id.speech_check_box).setVisibility(View.GONE);
        return rootView;
    }

    @Override
    protected void createDictionary() {
        File obbDir = activity.getObbDir();
        File f = new File(obbDir.getPath() + File.separator + "faces");
        faces = f.list();
    }

    @Override
    protected ArrayList backgroundArray() {
        ArrayList<Integer> indexList = new ArrayList<>(a.get(NO_OF_VALUES));
        ArrayList<Integer> arrayList = new ArrayList<>(a.get(NO_OF_VALUES));
        Random rand = new Random();
        int n;

        for (int i = 0; i < a.get(NO_OF_VALUES); i++) indexList.add(i);

        for (int i = 0; i < (a.get(NO_OF_VALUES)); i++) {
            n = rand.nextInt(indexList.size());
            arrayList.add(indexList.get(n));
            indexList.remove(n);
            if (a.get(RUNNING) == FALSE) break;
        }

        return arrayList;
    }
}

File: app/src/main/java/com/memory_athlete/memoryassistant/disciplines/Faces.kt
package com.memory_athlete.memoryassistant.disciplines

import android.annotation.SuppressLint
import android.content.Context
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.*
import com.memory_athlete.memoryassistant.R
import com.memory_athlete.memoryassistant.recall.RecallComplex
import com.squareup.picasso.Picasso
import java.io.File
import java.util.*
import kotlin.collections.ArrayList


class Faces : WordDisciplineFragment() {
    private lateinit var faces: Array<String>

    private val mFirstName = ArrayList<String>()
    private val mLastName = ArrayList<String>()

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        super.onCreateView(inflater, container, savedInstanceState)
        (rootView.findViewById<View>(R.id.no_of_values) as EditText).hint =
                getString(R.string.enter) + " " + getString(R.string.images)

        mRecallClass = RecallComplex::class.java
        hasSpeech = false
        rootView.findViewById<View>(R.id.speech_check_box).visibility = View.GONE
        return rootView
    }

    override fun createDictionary() {
        Names.readNames(mFirstName, mLastName, resources)
    }

    override fun backgroundArray(): ArrayList<*>? {
        val arrayList = ArrayList<RandomObject>(a[NO_OF_VALUES])
        val rand = Random()
        var n: Int
        var f: Int
        var l: Int

        for (i in 0 until a[NO_OF_VALUES]) {
            f = rand.nextInt(mFirstName.size)
            l = rand.nextInt(mLastName.size)
            // n = rand.nextInt()

            arrayList.add(RandomObject(f, l, i))

            if (a[RUNNING] == FALSE) break
        }

        return arrayList
    }

    @SuppressLint("InflateParams")
    override fun getMyView(convertView: View?, textSize: Int, item: Any?, context: Context?): View? {
        var linearLayout = convertView as LinearLayout?
        val randomItem = item as RandomObject
        if (linearLayout == null) {
            linearLayout = LayoutInflater.from(getContext())
                    .inflate(R.layout.item_category, null, true) as LinearLayout
            linearLayout.layoutParams = AbsListView.LayoutParams(ViewGroup.LayoutParams.MATCH_PARENT,
                    ViewGroup.LayoutParams.MATCH_PARENT)
        }

        val face = linearLayout.findViewById<ImageView>(R.id.face)
        val name = linearLayout.findViewById<TextView>(R.id.name)

        val s = mFirstName.get(randomItem.firstName) + " " + mLastName.get(randomItem.lastName)
        name.text = s

        Picasso
                //.setLoggingEnabled(true)
                .with(getContext())
                .load(activity.obbDir.path + File.separator + "faces" + File.separator + faces[randomItem.face])
                .placeholder(com.memory_athlete.memoryassistant.R.drawable.sa)
                .fit()
                //.centerInside()                 // or .centerCrop() to avoid a stretched imageÒ
                .into(face)

        return linearLayout
    }

    override fun startRandomAdapter(list: ArrayList<*>): RandomAdapter {
        return RandomAdapter(activity, list, 18)
    }
}
class RandomObject(val firstName: Int, val lastName: Int, val face: Int)